### PR TITLE
Fixed primary key violation issue in updating message id

### PR DESCRIPTION
**Pull request by AnujaLK**

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 1]
```diff
@@ -344,7 +345,11 @@ public void updateMessageID(String queueName, String nodeId, long startMessageId
                                 log.debug("RightExtra in overlapping slot : " + rightExtraSlot);
                             }
                             //Update last message ID - expand ongoing slot to cater this leftover part.
-                            slotAgent.addMessageId(queueName, lastMessageIdInTheSlot);
+                            try {
+                                slotAgent.addMessageId(queueName, lastMessageIdInTheSlot);
+                            } catch (AndesDataIntegrityViolationException ex) {
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: lets rename the vairable 'ignore'

Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: shouldn't this be handled in RDBMSAgent than passing it to this layer?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java [hunk 2]
```diff
@@ -159,7 +160,11 @@ public Slot getOverlappedSlot(String nodeId, String queueName) throws AndesExcep
 	 */
 	@Override
 	public void addMessageId(String queueName, long messageId) throws AndesException {
-		andesContextStore.addMessageId(queueName, messageId);
+		try {
+			andesContextStore.addMessageId(queueName, messageId);
+		} catch (AndesDataIntegrityViolationException ignore) {
+			//Primary key violation exception ignored
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java]: Please add a comment explaining why we do this.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java [hunk 3]
```diff
@@ -159,7 +160,11 @@ public Slot getOverlappedSlot(String nodeId, String queueName) throws AndesExcep
...
+		}
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java]: I am having a doubt whether this is indented properly.

**Comment by hemikak** [comment]
How was "Fixed primary key violation issue in updating message id" in this commit. Just asking.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java (hunk 2 above)
Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java]: Fixed with PR - https://github.com/wso2/andes/pull/202

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java (hunk 3 above)
Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java]: Fixed with PR - https://github.com/wso2/andes/pull/202

**Comment by AnujaLK** [comment]
When adding message id, same message id can be added to list when slots are overlapped. In RDBMS slot store composite primary key of queue name and message id have been used to avoid duplicates.
Therefore primary key violation exception is being thrown and can be ignored to fix the unnecessary log message console.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 4]
```diff
@@ -31,6 +31,7 @@
 import org.wso2.andes.server.cluster.coordination.SlotAgent;
 import org.wso2.andes.server.cluster.coordination.hazelcast.HazelcastAgent;
 import org.wso2.andes.server.cluster.coordination.rdbms.RDBMSAgent;
+import org.wso2.andes.store.AndesDataIntegrityViolationException;
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Do we need to import this? I can't find any usage.

Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: This is already fixed.

**Pull request merged**